File: BOJ/13458.cpp
#include <iostream>


using namespace std;



int N, A[1000000];
int B, C;


long long  solution() {
    long long cnt = 0;

    for(int i=0;i<N;i++){
        int res = A[i];
        res -= B;
        cnt++;
        
        if (res > 0) {
            cnt += res % C == 0 ? res / C : res / C + 1;
        }
    }

    return cnt;
}


int main() {
    freopen("input.txt","r",stdin);

    cin >> N;
    for (int i=0;i<N;i++){
        cin >> A[i];
    }
    cin >> B >> C;

    cout << solution();

}

File: BOJ/13458.py


import time

prev_time = time.time_ns()




import sys

# sys.stdin = open('input.txt','r')
read = sys.stdin.readline



def solution(A, B, C):
    cnt = 0

    for a in A:
        num = a - B
        cnt += 1
        if num > 0:
            cnt += num // C
            if num % C != 0:
                cnt += 1
    return cnt

N = int(read().strip())
A = list(map(int,read().split()))
B, C = map(int, read().split())

# print(N, A, B, C)

print(solution(A,B,C))









